inst/include/model_simulation.hpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include "types.hpp"
+
+namespace leapfrog {
+
+namespace internal {
+
+template<typename real_type>
+void run_hiv_model_simulation(int time_step,
+                              const Parameters<real_type> &pars,
+                              const State<real_type> &state_curr,
+                              State<real_type> &state_next,
+                              IntermediateData<real_type> &intermediate) {
+  run_add_new_hiv_infections(time_step, pars, state_curr, state_next, intermediate);
+
+  intermediate.everARTelig_idx =
+      pars.artcd4elig_idx(time_step) < pars.disease_stages ? pars.artcd4elig_idx(time_step) : pars.disease_stages;
+  for (int hiv_step = 0; hiv_step < pars.hiv_steps_per_year; ++hiv_step) {
+    run_disease_progression_and_mortality(time_step, pars, state_curr, state_next, intermediate, hiv_step);
+    run_new_infections(time_step, pars, state_curr, state_next, intermediate, hiv_step);
+  }
+}
+
+template<typename real_type>
+void run_add_new_hiv_infections(int time_step,
+                                const Parameters<real_type> &pars,
+                                const State<real_type> &state_curr,
+                                State<real_type> &state_next,
+                                IntermediateData<real_type> &intermediate) {
+  // TODO: Add EPP_DIRECTINCID switch
+
+  // Calculating new infections once per year (like Spectrum)
+  for (int g = 0; g < pars.num_genders; ++g) {
+    for (int a = pars.adult_incidence_first_age_group; a < pars.age_groups_pop; a++) {
+      intermediate.hiv_negative_pop(a, g) = state_curr.total_population(a, g) - state_curr.hiv_population(a, g);
+    }
+  }
+
+  for (int g = 0; g < pars.num_genders; g++) {
+    for (int a = pars.adult_incidence_first_age_group;
```

Comment:
because you have an a `a - pars.adult_incidence_first_age_group` later you could write this more simply 0..pars.pAG_INCIDPOP and do the addition where you need it

Reply:
Added a note to README for this